pkg/printer/workload_source_printer_test.go

```diff
@@ -183,3 +183,95 @@ branch:   my-branch
 		})
 	}
 }
+
+func TestWorkloadResourcesPrinter(t *testing.T) {
+	defaultNamespace := "default"
+	workloadName := "my-workload"
+
+	tests := []struct {
+		name           string
+		testWorkload   *cartov1alpha1.Workload
+		expectedOutput string
+	}{{
+		name: "various resources",
+		testWorkload: &cartov1alpha1.Workload{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      workloadName,
+				Namespace: defaultNamespace,
+			},
+			Status: cartov1alpha1.WorkloadStatus{
+				Resources: []cartov1alpha1.RealizedResource{
+					{
+						Name: "source-provider",
+						Conditions: []metav1.Condition{
```

Comment:
what is the desired behaviour if there is no `Ready` condition?  Also can you add a test with no `Ready` condition for a resource so we are aware of the impact on design